packages/image/src/colorExtractor.ts

```diff
@@ -0,0 +1,76 @@
+interface PaletteOptions {
+  colorCount?: number;
+  quality?: number;
+}
+
+/**
+ * Extracts a palette of colors from a given image URL based on provided options.
+ * @param {string} imageUrl - The URL of the image.
+ * @param {PaletteOptions} [options] - The extraction options.
+ */
+export const colorExtractor = (imageUrl: string, options?: PaletteOptions) => {
+  return loadImageFromUrl(imageUrl)
+    .then(extractImage)
+    .then((data) => {
+      if (!data) {
+        return [];
+      }
```

Comment:
I think extractColor don't need to load if user used load same url before.
So I want to suggest replacing it as parameter

```suggestion
export const extractColor = (loaded: ReturnType<typeof load>, options?: PaletteOptions) => {
  return extractColors(loaded)
    .then((data) => {
      if (!data) {
        return [];
      }
```

Reply:
no return in the `load` function.

Reply:
So It's suggestion how about make load return something

Reply:
good. This seems like it needs a little more thought. It seems like a meaningful review. thank you.